dojo/engagement/views.py

```diff
@@ -289,13 +290,16 @@ def delete_engagement(request, eid):
             form = DeleteEngagementForm(request.POST, instance=engagement)
             if form.is_valid():
                 product = engagement.product
-                engagement.delete()
-                messages.add_message(
-                    request,
-                    messages.SUCCESS,
-                    'Engagement and relationships removed.',
-                    extra_tags='alert-success')
-                create_notification(event='other',
+                if form.cleaned_data['asynchronous_delete']:
+                    dojo.tasks.asynch_delete.delay(Engagement, engagement.id)
+                else:
```

Comment:
Will this work if the checkbox is not ticked? I remember unticked checkboxes not being sent?
Probably a message should still be added saying the deletion was succesfully queued?
Please update the unit tests to reflect these changes.